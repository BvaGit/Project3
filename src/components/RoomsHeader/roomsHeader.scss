@import '../../utils/variables.scss';

.rooms-header{
    @include flex(column,_,_);
    padding: $padding_md;
    &__userinfo{
        @include flex(_,_,_);
    }
    &__userName{
        align-self: center;
        margin: 0 0 0 5px;
    }
    &__avatar{
        width: 40px;
        border: $border solid 1px;
        border-radius: 50%;
    }
    &__usermenu{
        @include flex(_,space-between,_);
    }
    &__settings{
        cursor: pointer;
        -webkit-transition: -webkit-transform 0.8s ease-in-out;
        transition: transform 0.8s ease-in-out;
    }
    &__settings:hover{
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
    &__logout{
        cursor: pointer;
        transition: transform 0.8s ease-in-out;
        margin: 0 0 0 5px;
    }
    &__logout:hover{
        -webkit-transform: scaleX(-1);
        transform: scaleX(-1);
    }
    &__search{
        margin: 0 auto;
        padding: $padding_md;
        width: 100%;
    }
    &__input{
        border: none;
        padding: $padding_md;
        background:rgb(226, 245, 253);
        width: 85%;
    }
    &__input::placeholder{
        color: $title-color;
        font-style: italic;
    }
}