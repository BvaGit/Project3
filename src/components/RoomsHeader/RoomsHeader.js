import React from "react";

import CustomModal from "../CustomModal";
import ModalMyAccount from "../ModalMyAccount";
import Settings from "../Settings";
import Input from "../Input";

import "./roomsHeader.scss";

const RoomsHeader = ({
  userLogin,
  userName,
  avatar,
  isOpen,
  modalMyAccount,
}) => {
  const handleClick = () => modalMyAccount();

  return (
    <div className="rooms-header">
      <div className="rooms-header__usermenu">
        <div className="rooms-header__userinfo">
          <img
            className="rooms-header__avatar"
            src={avatar ? avatar : "/assets/images/1-36-1024x1024.jpg"}
          />
          <div className="rooms-header__userName" onClick={handleClick}>
            {userName || userLogin}
          </div>
          {isOpen && (
            <CustomModal
              header="My Account"
              content={<ModalMyAccount handlecloseModal={handleClick} />}
              handlecloseModal={handleClick}
            />
          )}
        </div>
        <div className="rooms-header__settingsmenu">
          <Settings className="rooms-header__settings" />
          <img
            className="rooms-header__logout"
            src="/assets/images/logout_2.png"
          />
        </div>
      </div>
      <div className="rooms-header__search">
        <Input
          labelId="search"
          className="rooms-header__input"
          placeholderId="searchPlaceholder"
          titleId="Enter room name"
        />
      </div>
    </div>
  );
};
export default RoomsHeader;
