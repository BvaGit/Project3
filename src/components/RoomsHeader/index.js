import RoomsHeader from "./RoomsHeader";
import { languageValue } from "../../store/localization/selectors";
import { connect } from "react-redux";

const mapStateToProps = (state) => ({
    languageValue: languageValue(state),
})

export default connect(mapStateToProps, null)(RoomsHeader);