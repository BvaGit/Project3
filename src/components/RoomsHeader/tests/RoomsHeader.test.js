import React from 'react';
import { shallow, mount } from 'enzyme';
import RoomsHeader from '../RoomsHeader';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { IntlProvider } from "react-intl";

const mockStore = configureStore();
const store = mockStore(
    
);

describe('RoomsHeader', () => {
    it("Should match snapshot", () => {
        const component = shallowBiggerSmart(<RoomsHeader />, store)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(core);
};

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider>core</IntlProvider>);
};