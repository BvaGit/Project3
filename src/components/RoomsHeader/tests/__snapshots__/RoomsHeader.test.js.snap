// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomsHeader Should match snapshot 1`] = `"core"`;
